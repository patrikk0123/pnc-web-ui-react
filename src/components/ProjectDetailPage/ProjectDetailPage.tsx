import { Card, CardBody } from '@patternfly/react-core';
import { useCallback, useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { DataContainer } from '../../containers/DataContainer/DataContainer';
import { IService, useDataContainer } from '../../containers/DataContainer/useDataContainer';
import { useTitle } from '../../containers/useTitle';

import { projectService } from '../../services/projectService';

import { PageTitles } from '../../utils/PageTitles';

import { ActionButton } from '../ActionButton/ActionButton';
import { AttributesItems } from '../AttributesItems/AttributesItems';
import { SectionHeader } from '../SectionHeader/SectionHeader';
import { PageLayout } from './../PageLayout/PageLayout';

export const ProjectDetailPage = () => {
  const { projectId } = useParams();

  const dataContainer = useDataContainer(
    useCallback(({ requestConfig }: IService) => projectService.getProject({ id: projectId! }, requestConfig), [projectId])
  );
  const dataContainerRefresh = dataContainer.refresh;

  useTitle(`${dataContainer.data?.name} | ${PageTitles.projects}`);

  useEffect(() => dataContainerRefresh({ requestConfig: {} }), [dataContainerRefresh]);

  const attributes = [
    {
      name: 'Project URL',
      value: dataContainer.data?.projectUrl && (
        <a href={dataContainer.data?.projectUrl} target="_blank" rel="noopener noreferrer">
          {dataContainer.data?.projectUrl}
        </a>
      ),
    },
    {
      name: 'Issue Tracker URL',
      value: dataContainer.data?.issueTrackerUrl && (
        <a href={dataContainer.data?.issueTrackerUrl} target="_blank" rel="noopener noreferrer">
          {dataContainer.data?.issueTrackerUrl}
        </a>
      ),
    },
    { name: 'Engineering Team', value: dataContainer.data?.engineeringTeam },
    { name: 'Technical Leader', value: dataContainer.data?.technicalLeader },
  ];

  return (
    <DataContainer {...dataContainer} title="Project Details">
      <PageLayout title={dataContainer.data?.name} description={dataContainer.data?.description}>
        <SectionHeader
          actions={
            <ActionButton link="edit" iconType="edit">
              Edit
            </ActionButton>
          }
        />
        <div className="m-b-25">
          <Card>
            <CardBody>
              <AttributesItems attributes={attributes} />
            </CardBody>
          </Card>
        </div>
        <SectionHeader
          actions={
            <ActionButton action={() => console.log('Not implemented yet!')} iconType="create">
              Create
            </ActionButton>
          }
        >
          Build Configs
        </SectionHeader>
        <div className="m-b-25">
          <Card>
            <CardBody>
              <div style={{ backgroundColor: 'yellow', width: '100%', height: '30vh', textAlign: 'center' }}>
                TODO: Add Build Config table here and remove the style object
              </div>
            </CardBody>
          </Card>
        </div>
      </PageLayout>
    </DataContainer>
  );
};
