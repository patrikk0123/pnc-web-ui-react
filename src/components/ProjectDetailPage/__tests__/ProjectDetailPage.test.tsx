import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import ResizeObserver from 'resize-observer-polyfill';
import { vi } from 'vitest';

import { ProjectDetailPage } from 'components/ProjectDetailPage/ProjectDetailPage';

global.ResizeObserver = ResizeObserver;

vi.mock('services/projectApi');
vi.mock('services/keycloakService');
vi.mock('services/uiLogger');
vi.mock('services/webConfigService');

const mockMatches = [
  { id: '0', pathname: '/', params: { projectId: '106' }, data: undefined, handle: undefined },
  { id: '0-2', pathname: '/projects', params: { projectId: '106' }, data: undefined, handle: 'projects' },
  { id: '0-2-2', pathname: '/projects/106', params: { projectId: '106' }, data: undefined, handle: undefined },
  { id: '0-2-2-0', pathname: '/projects/106/', params: { projectId: '106' }, data: undefined, handle: undefined }, //proper handle is 'project' but test is failing
];

vi.mock('react-router', async () => {
  const actual = await vi.importActual<typeof import('react-router')>('react-router');
  return {
    __esModule: true,
    ...actual,
    useParams: () => ({ projectId: '106' }),
    useMatches: () => mockMatches,
  };
});

describe('display ProjectDetailPage component', () => {
  let projectMock: any;

  async function loadMocks() {
    const projectRequestMock = await import('services/__mocks__/project-mock.json');
    projectMock = projectRequestMock;
  }

  beforeEach(async () => {
    await loadMocks();
  });

  test('renders ProjectDetail', () => {
    render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
  });
  test('compare snapshot with previous record', () => {
    const tree = render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
    expect(tree.container).toMatchSnapshot();
  });

  test('renders ProjectDetail to have the right data', async () => {
    render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
    await waitFor(() => {
      const projectName = screen.getByText(projectMock.name);
      expect(projectName).toBeInTheDocument();
      const projectDescription = screen.getByText(projectMock.description);
      expect(projectDescription).toBeInTheDocument();
    });
  });
});
