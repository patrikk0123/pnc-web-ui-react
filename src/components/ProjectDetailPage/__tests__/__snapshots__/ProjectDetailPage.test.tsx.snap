// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display ProjectDetailPage component compare snapshot with previous record 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-breadcrumb"
      />
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content pull-left"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          />
          <p
            class=""
            data-ouia-component-id="OUIA-Generated-Text-5"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          />
        </div>
        <div
          class="pull-right display-flex align-items-center gap-10"
        >
          <div
            title="Keycloak service is not available."
          >
            <div
              class="disabled-content"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-primary pf-m-small"
                data-ouia-component-id="OUIA-Generated-Button-primary-3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                href="/edit"
              >
                Edit Project
              </a>
            </div>
          </div>
        </div>
      </section>
      <div
        class="pf-c-divider"
        role="separator"
      />
      <section
        class="pf-c-page__main-section"
      >
        <div
          class="shadow-global bg-w p-global margin-bottom"
        >
          <div
            class="responsive-content"
          />
        </div>
        <div
          class="shadow-global bg-w border-b-gray p-t-global p-b-global p-l-global p-r-global"
        >
          <div
            class=""
          >
            <div
              class="pf-l-flex pf-m-space-items-none"
              style="gap: 1.5rem;"
            >
              <div
                class=""
              >
                <div
                  class="pf-c-content"
                >
                  <h2
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-6"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    Build Configs
                  </h2>
                </div>
              </div>
              <div
                class=""
              >
                <div
                  title="Keycloak service is not available."
                >
                  <div
                    class="disabled-content"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary pf-m-small"
                      data-ouia-component-id="OUIA-Generated-Button-primary-4"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="/build-configs/create"
                    >
                      Create Build Config
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="shadow-global bg-w p-t-global p-b-global p-l-global p-r-global"
        >
          <div
            class=""
          >
            <div
              class="pf-l-flex pf-m-space-items-none"
              style="gap: 1.5rem;"
            >
              <div
                class=""
              >
                <div
                  class="pf-c-input-group"
                >
                  <div
                    class="pf-c-select"
                    data-ouia-component-id="OUIA-Generated-Select-single-2"
                    data-ouia-component-type="PF4/Select"
                    data-ouia-safe="true"
                    style="width: 200px;"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Options menu"
                      aria-labelledby=" pf-select-toggle-id-9"
                      class="pf-c-select__toggle"
                      id="pf-select-toggle-id-9"
                      type="button"
                    >
                      <div
                        class="pf-c-select__toggle-wrapper"
                      >
                        <span
                          class="pf-c-select__toggle-text"
                        >
                          Name
                        </span>
                      </div>
                      <span
                        class="pf-c-select__toggle-arrow"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <input
                    aria-invalid="false"
                    class="pf-c-form-control form-input"
                    data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                    data-ouia-component-type="PF4/TextInput"
                    data-ouia-safe="true"
                    id="filter-text"
                    placeholder="string | !string | s?ring | st*g"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="shadow-global bg-w border-t-gray"
        >
          <div
            class=""
          />
        </div>
        <div
          class="shadow-global bg-w"
        >
          <div
            class=""
          >
            <div>
              <div
                class="pf-c-pagination pf-m-bottom"
                data-ouia-component-id="OUIA-Generated-Pagination-bottom-2"
                data-ouia-component-type="PF4/Pagination"
                data-ouia-safe="true"
                id="options-menu-bottom-pagination"
                style="--pf-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
              >
                <div
                  class="pf-c-options-menu pf-m-top"
                  data-ouia-component-type="PF4/PaginationOptionsMenu"
                  data-ouia-safe="true"
                >
                  <div
                    class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                  >
                    <span
                      class="pf-c-options-menu__toggle-text"
                    >
                      <b>
                        0
                         - 
                        0
                      </b>
                       
                      of
                       
                      <b>
                        0
                      </b>
                       
                      
                    </span>
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Items per page"
                      class="  pf-c-options-menu__toggle-button"
                      data-ouia-component-id="OUIA-Generated-DropdownToggle-3"
                      data-ouia-component-type="PF4/DropdownToggle"
                      data-ouia-safe="true"
                      id="options-menu-bottom-toggle"
                      type="button"
                    >
                      <span
                        class="pf-c-options-menu__toggle-button-icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
                <nav
                  aria-label="Pagination"
                  class="pf-c-pagination__nav"
                >
                  <div
                    class="pf-c-pagination__nav-control pf-m-first"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to first page"
                      class="pf-c-button pf-m-plain pf-m-disabled"
                      data-action="first"
                      data-ouia-component-id="OUIA-Generated-Button-plain-5"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-c-pagination__nav-control"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to previous page"
                      class="pf-c-button pf-m-plain pf-m-disabled"
                      data-action="previous"
                      data-ouia-component-id="OUIA-Generated-Button-plain-6"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-c-pagination__nav-page-select"
                  >
                    <input
                      aria-label="Current page"
                      class="pf-c-form-control"
                      disabled=""
                      max="0"
                      min="1"
                      type="number"
                      value="0"
                    />
                    <span
                      aria-hidden="true"
                    >
                      of
                       
                      0
                    </span>
                  </div>
                  <div
                    class="pf-c-pagination__nav-control"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to next page"
                      class="pf-c-button pf-m-plain pf-m-disabled"
                      data-action="next"
                      data-ouia-component-id="OUIA-Generated-Button-plain-7"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-c-pagination__nav-control pf-m-last"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to last page"
                      class="pf-c-button pf-m-plain pf-m-disabled"
                      data-action="last"
                      data-ouia-component-id="OUIA-Generated-Button-plain-8"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                        />
                      </svg>
                    </button>
                  </div>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-breadcrumb"
    />
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content pull-left"
      >
        <h1
          class=""
          data-ouia-component-id="OUIA-Generated-Text-4"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        />
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-5"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        />
      </div>
      <div
        class="pull-right display-flex align-items-center gap-10"
      >
        <div
          title="Keycloak service is not available."
        >
          <div
            class="disabled-content"
          >
            <a
              aria-disabled="false"
              class="pf-c-button pf-m-primary pf-m-small"
              data-ouia-component-id="OUIA-Generated-Button-primary-3"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              href="/edit"
            >
              Edit Project
            </a>
          </div>
        </div>
      </div>
    </section>
    <div
      class="pf-c-divider"
      role="separator"
    />
    <section
      class="pf-c-page__main-section"
    >
      <div
        class="shadow-global bg-w p-global margin-bottom"
      >
        <div
          class="responsive-content"
        />
      </div>
      <div
        class="shadow-global bg-w border-b-gray p-t-global p-b-global p-l-global p-r-global"
      >
        <div
          class=""
        >
          <div
            class="pf-l-flex pf-m-space-items-none"
            style="gap: 1.5rem;"
          >
            <div
              class=""
            >
              <div
                class="pf-c-content"
              >
                <h2
                  class=""
                  data-ouia-component-id="OUIA-Generated-Text-6"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  Build Configs
                </h2>
              </div>
            </div>
            <div
              class=""
            >
              <div
                title="Keycloak service is not available."
              >
                <div
                  class="disabled-content"
                >
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary pf-m-small"
                    data-ouia-component-id="OUIA-Generated-Button-primary-4"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    href="/build-configs/create"
                  >
                    Create Build Config
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shadow-global bg-w p-t-global p-b-global p-l-global p-r-global"
      >
        <div
          class=""
        >
          <div
            class="pf-l-flex pf-m-space-items-none"
            style="gap: 1.5rem;"
          >
            <div
              class=""
            >
              <div
                class="pf-c-input-group"
              >
                <div
                  class="pf-c-select"
                  data-ouia-component-id="OUIA-Generated-Select-single-2"
                  data-ouia-component-type="PF4/Select"
                  data-ouia-safe="true"
                  style="width: 200px;"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-label="Options menu"
                    aria-labelledby=" pf-select-toggle-id-9"
                    class="pf-c-select__toggle"
                    id="pf-select-toggle-id-9"
                    type="button"
                  >
                    <div
                      class="pf-c-select__toggle-wrapper"
                    >
                      <span
                        class="pf-c-select__toggle-text"
                      >
                        Name
                      </span>
                    </div>
                    <span
                      class="pf-c-select__toggle-arrow"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <input
                  aria-invalid="false"
                  class="pf-c-form-control form-input"
                  data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                  data-ouia-component-type="PF4/TextInput"
                  data-ouia-safe="true"
                  id="filter-text"
                  placeholder="string | !string | s?ring | st*g"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shadow-global bg-w border-t-gray"
      >
        <div
          class=""
        />
      </div>
      <div
        class="shadow-global bg-w"
      >
        <div
          class=""
        >
          <div>
            <div
              class="pf-c-pagination pf-m-bottom"
              data-ouia-component-id="OUIA-Generated-Pagination-bottom-2"
              data-ouia-component-type="PF4/Pagination"
              data-ouia-safe="true"
              id="options-menu-bottom-pagination"
              style="--pf-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
            >
              <div
                class="pf-c-options-menu pf-m-top"
                data-ouia-component-type="PF4/PaginationOptionsMenu"
                data-ouia-safe="true"
              >
                <div
                  class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
                >
                  <span
                    class="pf-c-options-menu__toggle-text"
                  >
                    <b>
                      0
                       - 
                      0
                    </b>
                     
                    of
                     
                    <b>
                      0
                    </b>
                     
                    
                  </span>
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-label="Items per page"
                    class="  pf-c-options-menu__toggle-button"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-3"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe="true"
                    id="options-menu-bottom-toggle"
                    type="button"
                  >
                    <span
                      class="pf-c-options-menu__toggle-button-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <nav
                aria-label="Pagination"
                class="pf-c-pagination__nav"
              >
                <div
                  class="pf-c-pagination__nav-control pf-m-first"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to first page"
                    class="pf-c-button pf-m-plain pf-m-disabled"
                    data-action="first"
                    data-ouia-component-id="OUIA-Generated-Button-plain-5"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-c-pagination__nav-control"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to previous page"
                    class="pf-c-button pf-m-plain pf-m-disabled"
                    data-action="previous"
                    data-ouia-component-id="OUIA-Generated-Button-plain-6"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-c-pagination__nav-page-select"
                >
                  <input
                    aria-label="Current page"
                    class="pf-c-form-control"
                    disabled=""
                    max="0"
                    min="1"
                    type="number"
                    value="0"
                  />
                  <span
                    aria-hidden="true"
                  >
                    of
                     
                    0
                  </span>
                </div>
                <div
                  class="pf-c-pagination__nav-control"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to next page"
                    class="pf-c-button pf-m-plain pf-m-disabled"
                    data-action="next"
                    data-ouia-component-id="OUIA-Generated-Button-plain-7"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-c-pagination__nav-control pf-m-last"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to last page"
                    class="pf-c-button pf-m-plain pf-m-disabled"
                    data-action="last"
                    data-ouia-component-id="OUIA-Generated-Button-plain-8"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                      />
                    </svg>
                  </button>
                </div>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
